packages/shared/src/parsing/urls.ts

```
@@ -0,0 +1,26 @@
+export function parseURLsFromString(input: string): URL[] {
+  const matches = input.match(urlRegex)
+  if (!matches) return []
+  return matches.map((match) => createURL(match)).filter((url) => url !== undefined)
+}
+
+export function isIPAddress(input: URL | string): boolean {
+  const url = createURL(input)
+  if (!url) return false
+
+  const hostname = url.hostname
+  const isLikeIPV4 = hostname.split('.').every((part) => !isNaN(Number(part)))
```

Biome

[error] isNaN is unsafe. It attempts a type coercion. Use Number.isNaN instead.

See the MDN documentation for more details.
Unsafe fix: Use Number.isNaN instead.


(lint/suspicious/noGlobalIsNan)